### PR TITLE
Fix password reset to random password instead of empty password

### PR DESCRIPTION
**Pull request by tigusoft-vm**
See issue [#23509](https://mantisbt.org/bugs/view.php?id=23509)

### CODE REVIEW

File: lang/strings_polish.txt [hunk 1]
```diff
@@ -726,6 +726,7 @@ $s_reset_password_button = 'Nowe hasło';
 $s_delete_user_button = 'Usuń Użytkownika';
```
Comment (atrol) [inline lang/strings_polish.txt]: This is not the way we change on English messages.
Translations are managed via TranslateWiki. Please refer our own Wiki [1] for the process to submit new translations or revise existing ones.

[1] http://www.mantisbt.org/wiki/doku.php/mantisbt:translationshowto

File: manage_user_edit_page.php [hunk 2]
```diff
@@ -227,7 +227,10 @@
 	if( ( ON == config_get( 'send_reset_password' ) ) && ( ON == config_get( 'enable_email_notification' ) ) ) {
 		echo lang_get( 'reset_password_msg' );
 	} else {
-		echo lang_get( 'reset_password_msg2' );
+		$msg = lang_get( 'reset_password_msg3' );
+		# if translation for random-password is missing, then create a warning based on other translated words
```
Comment (atrol) [inline manage_user_edit_page.php]: This is not needed as we fallback to English if a translation is missing.

**Comment by atrol** [comment]
Thanks for your contribution.

BTW, just got aware of 
https://www.mantisbt.org/bugs/view.php?id=15658
https://github.com/mantisbt-plugins/AdminSetPassword

**Comment by cproensa** [comment]
There is not reference to a bugtracker issue.

Also, In this situation, some applications allows to set an initial password, which can be used only once, and then prompts the user to change to a personal one.
This solves this same scenario where email is not suitable, and adds a layer of security by keeping the user password private (this may be a requeriment by some security policies)

File: manage_user_reset.php [hunk 3]
```diff
@@ -85,7 +97,12 @@
 		echo lang_get( 'account_reset_protected_msg' );
 	} else {
 		# SUCCESSFUL RESET
-		if( ( ON == config_get( 'send_reset_password' ) ) && ( ON == config_get( 'enable_email_notification' ) ) ) {
+		if( $password_kind === 'random_set' ) {
+			# reseted it as random text shown here directly
+			echo lang_get( 'lost_password_title' ).': ';
+			echo lang_get( 'password' ).': '.'<b>'.htmlspecialchars($password_created).'</b>';
+		}
+		else if( ( ON == config_get( 'send_reset_password' ) ) && ( ON == config_get( 'enable_email_notification' ) ) ) {
```
Comment (k1931547) [inline manage_user_reset.php]: true

**Comment by l266509** [comment]
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  Whats status of this PR?

**Comment by dregad** [comment]
> There is not reference to a bugtracker issue.

I just opened [#23509](https://mantisbt.org/bugs/view.php?id=23509).

**Comment by vboctor** [comment]
Closing this PR due to lack of activity.  Feel free to re-open or submit a new one.

**Pull request closed without merging**